### ISSUE TITLE
Browser MSSQL add connection dialogue doesn't allow DSN without entering a Host name

### ISSUE THREAD

**Issue by Geojim** [qgis/QGIS#38197]
Using the left pane browser, MSSQL to add a connection, the OK button is greyed out until a host is added. If using a DSN a host shouldn't be needed.

QGIS 14.1
W10-64

This seems like a regressive change, pretty sure it was working in 3.12(?)

**Comment by agiudiceandrea** [comment]
Hi @Geojim, is it similar to issue https://github.com/qgis/QGIS/issues/37427 already reported and fixed with PR https://github.com/qgis/QGIS/pull/37467 by @nyalldawson?

So, it seems the logic in 
https://github.com/qgis/QGIS/blob/2b3b88c51524814a839a8259e7575868a0e81714/src/providers/mssql/qgsmssqlnewconnection.cpp#L252-L257

should be (if it is allowed to insert both host and Provider/DSN parameter):
```c++
bool enabled = !txtName->text().isEmpty() && !( txtService->text().isEmpty() && txtHost->text().isEmpty() ) && item;
```

or (if it isn't allowed to insert both host and Provider/DSN parameters):
```c++
bool enabled = !txtName->text().isEmpty() && ( txtService->text().isEmpty() == !txtHost->text().isEmpty() ) && item;
```

**Comment by Geojim** [comment]
Very similar issue to #37427, but should be an and/or test for the Service and Host fields.

I'll guess at the right code.. (sorry)
`bool enabled = !txtName->text.isEmpty && !(txtService->text().isEmpty() || txtHost->text().isEmpty() ) && item;`

**Comment by Geojim** [comment]
Same story in 3.14.15...

**Comment by agiudiceandrea** [comment]
Hi @Geojim, I was hoping some skilled dev would addressed the issue. I've just submitted a Pull Request to raise the attention on this bug and try to fix it.

**Comment by agiudiceandrea** [comment]
Hi @Geojim, if you want you could test the patch: just download the "QGIS for Windows 64bit" artifact (QGIS for Windows 64bit.zip) from https://github.com/qgis/QGIS/actions/runs/236224903, extract the contained qgis-mxe-release.zip file, extract the content of qgis-mxe-release.zip and then run qgis.exe in the "qgis-mxe-release" folder. Please let us know if that's okay with you.

**Comment by Geojim** [comment]
Thanks @agiudiceandrea for sorting out. 

Sorry, but quite difficult for me to test unfortunately. I don't have admin access to fiddle with patches at work, which is the environment I'd need to be testing out the DSN functionality etc....

**Comment by agiudiceandrea** [comment]
Ok @Geojim, the fix will be also available in the upcoming QGIS 3.14.16 Latest release.
